cli/cli_connect_test.go

```diff
@@ -18,39 +20,92 @@ import (
 	"google.golang.org/grpc"
 )
 
+func captureOutput(f func()) (string, error) {
+	reader, writer, err := os.Pipe()
+	if err != nil {
+		return "", err
+	}
+	stdout := os.Stdout
+	stderr := os.Stderr
+	defer func() {
+		os.Stdout = stdout
+		os.Stderr = stderr
+	}()
+	os.Stdout = writer
+	os.Stderr = writer
+	out := make(chan string)
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		var buf bytes.Buffer
+		wg.Done()
+		io.Copy(&buf, reader)
+		out <- buf.String()
+	}()
+	wg.Wait()
+	f()
+	writer.Close()
```

Comment:
Why `reader` isn't closed? And should probably just do at the top:
```
	reader, writer, err := os.Pipe()
	if err != nil {
		return "", err
	}
    defer reader.Close()
    defer writer.Close()
```

Reply:
the `writer` needs to be closed before to unlock `io.Copy` from `reader`.

by default when a `pipe` is created it will add a cleanup function when garbage collector releases the memory, so reader will be closed by GC: `runtime.SetFinalizer(f.file, (*file).close)` from https://github.com/golang/go/blob/release-branch.go1.20/src/os/file_unix.go#L215